### PR TITLE
ISPN-9286 WriteBehindFaultToleranceTest.testBlockingOnStoreAvailabili…

### PR DESCRIPTION
**Pull request by ryanemerson**
…tyChange fix

https://issues.jboss.org/browse/ISPN-9286

### CODE REVIEW

File: core/src/test/java/org/infinispan/persistence/WriteBehindFaultToleranceTest.java [hunk 1]
```diff
@@ -60,11 +60,10 @@ public void testBlockingOnStoreAvailabilityChange() {
       store.setAvailable(false);
       assertFalse(store.isAvailable());
 
-      TestingUtil.sleepThread(AVAILABILITY_INTERVAL * 5);
       // PM & AsyncWriter should still be available as the async modification queue is not full
       assertTrue(asyncWriter.isAvailable());
       assertFalse(TestingUtil.extractField(asyncWriter, "delegateAvailable"));
-      assertTrue(pm.isAvailable());
+      eventually(pm::isAvailable);
```
Comment (danberindei) [inline core/src/test/java/org/infinispan/persistence/WriteBehindFaultToleranceTest.java]: Doesn't the fact that we have to wait for the PM to become available contradict the comment above?

Comment (ryanemerson) [reply core/src/test/java/org/infinispan/persistence/WriteBehindFaultToleranceTest.java]: Yes you're right. I was obviously in auto-pilot when I changed this line.

**Comment by danberindei** [comment]
@ryanemerson There are 2 more instances of `TestingUtil.sleepThread(AVAILABILITY_INTERVAL * 5)`, and sleeping for 50ms really doesn't guarantee anything in CI :)

**Comment by ryanemerson** [comment]
@danberindei Good point. I've updated the test to remove the remaining offenders.

File: core/src/test/java/org/infinispan/persistence/WriteBehindFaultToleranceTest.java [hunk 2]
```diff
@@ -49,43 +51,42 @@
...
 
       // Add entries >= modification queue size so that store is no longer available
       cache.putAll(intMap(0, 10));
       assertEquals(1, store.size());
 
-      TestingUtil.sleepThread(AVAILABILITY_INTERVAL * 5);
       // PM and writer should not be available as the async modification queue is now oversubscribed and the delegate is still unavailable
       assertFalse(asyncWriter.isAvailable());
-      assertFalse(pm.isAvailable());
+      eventually(() -> !pm.isAvailable());
```
Comment (danberindei) [inline core/src/test/java/org/infinispan/persistence/WriteBehindFaultToleranceTest.java]: Sorry, but I don't think this is right: IMO the new `eventually` calls should be exactly where the `sleepThread` calls were

Comment (ryanemerson) [reply core/src/test/java/org/infinispan/persistence/WriteBehindFaultToleranceTest.java]: done

Comment (danberindei) [reply core/src/test/java/org/infinispan/persistence/WriteBehindFaultToleranceTest.java]: @ryanemerson I should have been more clear: I was thinking about all the new `eventually` calls (except for `eventually(() -> 10 == store.size());`, that one's fine where it is) :)

Comment (ryanemerson) [reply core/src/test/java/org/infinispan/persistence/WriteBehindFaultToleranceTest.java]: @danberindei I think that's what I have done, except one of the `Thread.sleep` calls was in fact redundant so I don't need to replace with an eventually call.

Comment (danberindei) [reply core/src/test/java/org/infinispan/persistence/WriteBehindFaultToleranceTest.java]: Ok, when I looked it seemed like you only changed this instance. Merging...

**Comment by danberindei** [comment]
Integrated, thanks Ryan!

**Pull request merged**